project/src/main/scala/ScapegoatInspectionsGenerator.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+import java.nio.file.{Path, Paths}
+import com.sksamuel.scapegoat.Inspection
+import io.github.classgraph.ClassGraph
+import sbt.Keys._
+import sbt._
+
+import scala.meta._
+import scala.collection.JavaConverters._
+
+/** SBT Task that generates a managed file with all scapegoat inspections */
+object ScapegoatInspectionsGenerator {
+
+  /** Scapegoat inspections that won't be included in the generated file */
+  val BlacklistedInspections = Set(
+    "com.sksamuel.scapegoat.inspections.collections.FilterDotSizeComparison", // Not implemented yet
+    "com.sksamuel.scapegoat.inspections.collections.ListTail" // Not implemented yet
+  )
+
+  val generatorTask = Def.task {
+    val log = streams.value.log
+    log.info("Generating the scapegoat inspections file.")
+
+    // Load the template file.
+    val templateFile = Paths
+      .get(
+        baseDirectory.value.toString,
+        "project",
+        "src",
+        "main",
+        "resources",
+        "ScapegoatInspections.scala"
+      )
+
+    val allScapegoatInspections = extractInspections()
+    val stringifiedScapegoatIsnpections = stringifyInspections(allScapegoatInspections)
+    val transformed = fillTemplate(templateFile, stringifiedScapegoatIsnpections)
+
+    val scapegoatInspectionsFile = (sourceManaged in Compile).value / "scapegoat" / "inspections.scala"
+    IO.write(scapegoatInspectionsFile, transformed)
+
+    Seq(scapegoatInspectionsFile)
+  }
+
+  /** Returns all scapegoat inspections, except the ones that should be ignored */
+  def extractInspections(): List[(String, Inspection)] =
```

Comment:
Do you have any ideas for testing this method?

Reply:
🤔 I guess to test `ClassGraph` in isolation you would need to call this function and inspect the collection that you get back. IMHO it's probably not necessary since your other tests already cover this in a way (indirectly).